process/block/postprocess/basePostProcess.go

```diff
@@ -171,3 +179,54 @@ func (bpp *basePostProcessor) RemoveProcessedResultsFor(txHashes [][]byte) {
 		delete(bpp.mapTxToResult, string(txHash))
 	}
 }
+
+func (bpp *basePostProcessor) splitMiniBlocksIfNeeded(miniBlocks []*block.MiniBlock) []*block.MiniBlock {
+	splitMiniBlocks := make([]*block.MiniBlock, 0)
+	mapGasLimitInReceiverShard := make(map[uint32]uint64)
+
+	for _, miniBlock := range miniBlocks {
+		receiverShardID := miniBlock.ReceiverShardID
+		currentMiniBlock := createEmptyMiniBlock(miniBlock)
+		mapGasLimitInReceiverShard[receiverShardID] = 0
+
+		for _, txHash := range miniBlock.TxHashes {
+			interResult, ok := bpp.interResultsForBlock[string(txHash)]
+			if !ok {
+				log.Warn("basePostProcessor.splitMiniBlocksIfNeeded: missing tx", "hash", txHash)
```

Comment:
this should be a critical bug, not ?

Reply:
Yes, this will be caught up in VerifyInterMiniBlocks when the hash of the each created mbs will not matched with the hash of each received mini block. Because CreateAllInterMiniBlocks does not return error, and this method is called from there, it would have been useless to return error from here just to log it there

Reply:
ok. can remain